### PR TITLE
Add password map file support for redis_exporter

### PR DESCRIPTION
**Pull request by spartan0x117**
#### PR Description
Adds support for specifying a `redis_password_map_file` in `redis_exporter`, which consists of JSON KV pairs mapping redis uris to passwords. 

#### PR Checklist

- [X] CHANGELOG updated
- [ ] Documentation added
- [X] Tests updated

### CODE REVIEW

File: pkg/integrations/redis_exporter/redis_exporter.go [hunk 1]
```diff
@@ -41,6 +41,7 @@ type Config struct {
 	RedisUser               string             `yaml:"redis_user,omitempty"`
 	RedisPassword           config_util.Secret `yaml:"redis_password,omitempty"`
 	RedisPasswordFile       string             `yaml:"redis_password_file,omitempty"`
+	RedisPasswordMapFile    string             `yaml:"redis_password_map_file,omitempty"`
```
Comment (rfratto) [inline pkg/integrations/redis_exporter/redis_exporter.go:44]: WDYT about making it be invalid to set more than one of these at a time?

Comment (spartan0x117) [reply pkg/integrations/redis_exporter/redis_exporter.go:44]: I think that would make things clearer, I had just been deferring to the downstream

File: docs/sources/configuration/integrations/redis-exporter-config.md [hunk 2]
```diff
@@ -81,6 +81,9 @@ Full reference of options:
   # over redis_password.
   [redis_password_file: <string>]
 
+  # Path of a file containing a json definiton of redis uri strings to passwords (e.g. {"redis://localhost:6379": "sample_password"})
```
[the remark this answers is not in the document]
Comment (karengermond) [reply docs/sources/configuration/integrations/redis-exporter-config.md:84]: I'm also not sure what "to passwords" means here. Should it  be:

Path of a file containing a JSON definition of Redis URI strings for passwords (e.g. {"redis://localhost:6379": "sample_password"})

**Review by karengermond — APPROVED** [review]
LGTM, minor feedback.

File: docs/sources/configuration/integrations/redis-exporter-config.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (rfratto) [reply docs/sources/configuration/integrations/redis-exporter-config.md:84]: I think the intent was for this to be something like:

> ```Path of a JSON file which containing an object to map Redis URIs to passwords```

[the remark this answers is not in the document]
Comment (spartan0x117) [reply docs/sources/configuration/integrations/redis-exporter-config.md:84]: Ah yes, I wanted to convey that the password file contained a flat JSON structure which maps Redis URI -> password (ultimately both are strings)

**Review by rfratto — APPROVED** [review]
Thanks!

**Pull request merged**